backend/scripts/scraping.py

```diff
@@ -0,0 +1,319 @@
+from bs4 import BeautifulSoup
+import re
+
+
+def get_dict_courses(url):
+    """Get class_ids and classes' names from ensae.fr
+
+    Args:
+        url (_type_): url for scraping
+    """
+    response = requests.get(url)
+    courses_dict = {}
+
+    if response.status_code == 200:
+        soup = BeautifulSoup(response.text, 'html.parser') # Parse the HTML content of the page
+        links = soup.find_all('a') # Extract data by navigating the HTML structure
+
+        for link in links:
+            if link.get('href').startswith('/courses'):
+                #get names
+                course_name = link.get_text(strip=True)
+                course_name = re.sub(r'^ou', '', course_name) #some names have "ou" at the beginning - remove "ou"
+                course_name = re.sub(r'\s*\([^)]*\)', '', course_name) #some names have () with details - remove (...)
+
+                #get id
+                href = link.get('href')
+                course_id = href.split('/courses/')[-1]
+                numbers = re.findall(r'\d+', course_id)[0] #some ids contain str - remove str
+
+                #stock courses in dict
+                courses_dict[course_name] = numbers
+
+    else:
+        print('Failed to retrieve the webpage')
+    
+    return(courses_dict)
+
+#courses url
+url_1a = "https://www.ensae.fr/formation/cycle-ingenieur/premiere-annee-du-cycle-ingenieur"
+url_2a = "https://www.ensae.fr/formation/cycle-ingenieur/deuxieme-annee"
+url_3a = "https://www.ensae.fr/formation/cycle-ingenieur/troisieme-annee/catalogue-des-cours-de-troisieme-annee-du-cycle-ingenieur"
+
+courses_ensae_dict = {key: value for d in (get_dict_courses(url_1a), get_dict_courses(url_2a), get_dict_courses(url_3a)) for key, value in d.items()}
+
+
+
+def transform_date_french_to_iso(date_french):
+    # Split the date string into its components
+    day, month, year = date_french.split()
+    
+    # Define a mapping of French month names to month numbers
+    month_mapping = {
+        "janvier": "01",
+        "février": "02",
+        "mars": "03",
+        "avril": "04",
+        "mai": "05",
+        "juin": "06",
+        "juillet": "07",
+        "août": "08",
+        "septembre": "09",
+        "octobre": "10",
+        "novembre": "11",
+        "décembre": "12"
+    }
+
+    # Format the date components in the desired format
+    formatted_date = f"{year}-{month_mapping[month.lower()]}-{day.zfill(2)}"
+
+    return formatted_date
+
+
+from datetime import datetime
+def transform_time(original_time):
+    try:
+        # Parse the original time using the original format
+        original_datetime = datetime.strptime(original_time, '%H:%M')
+
+        # Format the time to match the new format '%H:%M:%S'
+        new_time_format = original_datetime.strftime('%H:%M:%S')
+
+        return new_time_format
+    except ValueError:
+        # Handle the case where the input time format is invalid
+        return "Invalid input time format"
```

Comment:
Might as well raise an error, so that the string doesn't get used by another function

Reply:
(instead of returning it)